gammapy/maps/wcsnd.py

```diff
@@ -460,51 +460,150 @@ def downsample(self, factor, preserve_counts=True):
             data /= factor**2
         return self.__class__(geom, data, meta=copy.deepcopy(self.meta))
 
-    def plot(self, ax=None, idx=None, **kwargs):
-        """Quickplot method.
+
+
+    
+    def plot(self, ax=None, idx=None, fig=None, add_cbar=False, stretch='linear', smooth=None, radius=1, **kwargs):
+        """
+        Plot image on matplotlib WCS axes
 
         Parameters
         ----------
-        norm : str
-            Set the normalization scheme of the color map.
+        ax : `~astropy.visualization.wcsaxes.WCSAxes`, optional
+            WCS axis object to plot on.
         idx : int or tuple
             Set the image slice to plot if this map has non-spatial dimensions.
             For maps with exactly one non-spatial dimension idx can be an int
-        **kwargs : dict
-            Keyword arguments passed to `~matplotlib.pyplot.imshow`.   
+        TODO: let idx work with slicing
+
+        fig : `~matplotlib.figure.Figure`, optional
+            Figure
+        stretch : str, optional
+            Scaling for image ('linear', 'sqrt', 'log').
+            Similar to normalize and stretch functions in ds9.
+            See http://docs.astropy.org/en/stable/visualization/normalization.html
+
+        smooth: ~string : either 'gauss' or 'disk' 
+            kernel shape for smoothing the image.  
+
+            radius:  `~astropy.units.Quantity` or float
+            
+            Smoothing width given as quantity or float. If a float is given it
+            interpreted as smoothing width in pixels. If an (angular) quantity
+            is given it converted to pixels using `geom.wcs.wcs.cdelt`.
+            Default smoothing is of 1 pixel (no smooth)
+
+        
+            The definition of the smoothing parameter radius is equivalent to the
+            one that is used in ds9 (see `ds9 smoothing <http://ds9.si.edu/doc/ref/how.html#Smoothing>`_).
+        
+        kwargs: for parameters to pass into imshow
 
         Returns
         -------
         fig : `~matplotlib.figure.Figure`
-            Figure object.
+            Figure
         ax : `~astropy.visualization.wcsaxes.WCSAxes`
             WCS axis object
-        im : `~matplotlib.image.AxesImage`
-            Image object.
+        cbar : `~matplotlib.colorbar.Colorbar`
+            Colorbar object (if ``add_cbar=True`` was set, else ``None``)
+
         """
+        
         import matplotlib.pyplot as plt
-        import matplotlib.colors as colors
-
-        if ax is None:
+        from astropy.visualization import simple_norm
+        from scipy.ndimage import gaussian_filter
+        from scipy.signal import convolve
+        from scipy.stats import gmean
+        from astropy.convolution import Tophat2DKernel
+        from astropy.coordinates import Angle
```

Comment:
Please move all imports to the top, except for the optional dependencies such as `scipy`.